lab-bundle-spec.md

```diff
@@ -158,21 +153,35 @@ environment_resources:
 
 ##### GCP Project (gcp_project)
 
-attribute                     | required | type    | notes
------------------------------ | -------- | ------- | --------------------------------------
-dm_template.script            |          | path    | Relative path to a Deployment Manager directory tree.
-dm_template.custom_properties |          | array   | Array of key/value pairs.
-fleet                         |          | enum*   | Specify a Qwiklabs fleet to pull the project from.
+attribute                           | required | type    | notes
+----------------------------------- | -------- | ------- | --------------------------------------
+dm_template.start.script            |          | path    | Relative path to a Deployment Manager directory tree.
```

Comment:
this hierarchy (dm_template -> start / end -> script / custom_properties) feels a little awkward to me

Reply:
would you prefer something like
```
startup_template:
  script: ...
  custom_properties: ...
cleanup_template:
  script: ...
  custom_properties: ...
```
?

Reply:
I do prefer the latter.

`startup_script` and `cleanup_script` as top-level/first-class properties emphasize the role it plays on the lab life-cycle. The fact that it uses deployment manager is an implementation detail that could change.

Note: not wedded to "script" as the noun but I prefer it to "template")

Note: My argument assumes that the clean-up and start-up scripts actually work independently of each other (e.g. don't rely on a shared DM state)... I hope this is the case, but I might be missing a key implementation concern.

Reply:
We should indicate the type of script somewhere. Whether that's expressed a `type` field or a nested object, I'm agnostic.